hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieAppendHandle.java

```diff
@@ -484,7 +485,7 @@ protected void doWrite(HoodieRecord record, Schema schema, TypedProperties props
     } catch (Throwable t) {
       // Not throwing exception from here, since we don't want to fail the entire job
       // for a single record
-      writeStatus.markFailure(record, t, recordMetadata);
+      writeStatus.markFailure(lazily(() -> HoodieRecordDelegate.fromHoodieRecord(record)), t, recordMetadata);
```

Comment:
What the purpose to make it lazy, the closure already includes the `HoodieRecord`.

Reply:
I mean the closure already incudes a `HudiRecord`, which is much bigger than the `HoodieRecordDelegate` object iftself.

Reply:
correct. that's what i meant by minor saving as the delegate is lean